File: aetherboard/unity/AetherboardVR/Assets/Aetherboard/Scripts/Core/CoopRules.cs
namespace Aetherboard.Core
{
    /// <summary>
    /// Split-coop ownership: P1 = 铁卫/游弦, P2 = 白愈/黑炎.
    /// </summary>
    public static class CoopRules
    {
        public static readonly string[] Player1Units = { "knight", "bard" };
        public static readonly string[] Player2Units = { "white_mage", "black_mage" };

        public static bool CanControl(int playerId, string unitId, bool coopEnabled)
        {
            if (!coopEnabled || playerId <= 0 || string.IsNullOrEmpty(unitId)) return true;
            var allowed = playerId == 1 ? Player1Units : Player2Units;
            foreach (var id in allowed)
                if (id == unitId) return true;
            return false;
        }

        public static bool CommandRequiresUnit(BattleCommandType type) =>
            type is BattleCommandType.Move or BattleCommandType.Skill;

        public static int OwnerOf(string unitId)
        {
            foreach (var id in Player1Units)
                if (id == unitId) return 1;
            foreach (var id in Player2Units)
                if (id == unitId) return 2;
            return 0;
        }
    }
}

File: aetherboard/sim/coop_rules.py
"""Split-coop unit ownership rules."""

from __future__ import annotations

PLAYER1_UNITS = frozenset({"knight", "bard"})
PLAYER2_UNITS = frozenset({"white_mage", "black_mage"})


def can_control(player_id: int, unit_id: str, coop_enabled: bool) -> bool:
    if not coop_enabled or player_id <= 0 or not unit_id:
        return True
    allowed = PLAYER1_UNITS if player_id == 1 else PLAYER2_UNITS
    return unit_id in allowed


def command_requires_unit(cmd_type: str) -> bool:
    return cmd_type in {"Move", "Skill"}
